to-do-list/src/List.js

```diff
@@ -0,0 +1,26 @@
+import "./List.css";
+
+const ToDoList = () => {
+  return (
+    <div className="to-do-list">
+      <header>To Do List</header>
+      <input id="input-item" type="text"></input>
+      <button onClick={e => addItem(e.view.document)}>Add To-Do</button>
+      <div>
+        <ul id="list"></ul>
+      </div>
+    </div>
+  );
+};
+
+const addItem = props => {
+  let list = props.getElementById("list");
+  let listItem = props.createElement("li");
+  let text = props.getElementById("input-item").value;
+
+  listItem.appendChild(document.createTextNode(text));
+  list.appendChild(listItem);
+  props.getElementById("input-item").value = "";
+};
```

Comment:
That's the correct way of adding an item to a list using JS. To make it more optimized, instead of creating and appending elements every time, you can have a state variable that keeps track of all the items in the list. So when you want to add something, you can add that value to the state variable, and then the file `ToDoList.js` will take care of displaying the newly added item.

Let me explain by giving a small example:
```
constructor(props) {
    super(props);
    this.state = {
      singleItem: '', //item that the user adds
      items: [] //all the items
    };
  }

handleChange = (event) => {
    this.setState({ singleItem: event.target.value });
  }

onSubmit = (event) => {
    event.preventDefault();
    this.setState({
      singleItem: '',
      items: [...this.state.items, this.state.singleItem]
    });
  }


<form onSubmit={this.onSubmit}>
    <input value={newItem} type="text" onChange={this.handleChange} />
    <button>Submit</button>
</form>
<List items={this.state.items} />
```


And then in your `ToDoList.js` file, you can display the entire list with something like this:
```


const List = props => (
  <ul>
    {
      props.items.map((item, index) => <li key={index}>{item}</li>)
    }
  </ul>
);
```